src/string_immutable.zig

```diff
@@ -5905,51 +5886,84 @@ pub const visible = struct {
         return len;
     }
 
-    fn visibleUTF16WidthFn(input: []const u16, comptime asciiFn: anytype) usize {
-        var bytes = input;
+    fn visibleUTF16WidthFn(input_: []const u16, exclude_ansi_colors: bool) usize {
+        var input = input_;
+        _ = &input;
         var len: usize = 0;
-        while (bun.strings.firstNonASCII16CheckMin([]const u16, bytes, false)) |i| {
-            len += asciiFn(bytes[0..i]);
-            bytes = bytes[i..];
 
-            const utf8 = utf16CodepointWithFFFD([]const u16, bytes);
-            len += visibleCodepointWidthType(u32, utf8.code_point);
-            bytes = bytes[@min(@as(usize, utf8.len), bytes.len)..];
+        if (firstNonASCII16([]const u16, input)) |idx| {
+            for (0..idx) |i| {
+                len += visibleLatin1WidthScalar(@intCast(input[i]));
+            }
+            input = input[idx..];
         }
 
-        len += asciiFn(bytes);
-
+        var i: usize = 0;
+        var iter = std.unicode.Utf16LeIterator.init(input);
```

Comment:
We should not use std.unicode.Utf16LeIterator. 

This function can be used instead, and the code will get simpler.
```zig
const utf8 = utf16CodepointWithFFFD([]const u16, bytes);
```